packages/synapse-react-client/src/components/TermsAndConditions/TermsAndConditionsItem.tsx

```diff
@@ -19,6 +28,43 @@ const TermsAndConditionsItem: React.FunctionComponent<
   TermsAndConditionsItemProps
 > = props => {
   const { id, item, enabled, checked, onChange } = props
+  const { iconFileHandleId, label, description } = item
+  const [iconFileResult, setIconFileResult] = useState<FileResult | undefined>(
+    undefined,
+  )
+  const [iconFileContent, setIconFileContent] = useState<string | undefined>(
+    undefined,
+  )
+
+  useEffect(() => {
```

Comment:
2 step process here.  First we get the presigned URL and file handle associated to this file handle ID, and save that fileResult in the state

Reply:
It's close!  I'll add a new FileHandleAssociation-based hook